src/wallet/spend.cpp

```diff
@@ -516,14 +516,32 @@ std::optional<SelectionResult> ChooseSelectionResult(const CWallet& wallet, cons
         results.push_back(*srd_result);
     }
 
-    if (results.size() == 0) {
+    if (results.empty()) {
         // No solution found
         return std::nullopt;
     }
 
+    std::vector<SelectionResult> eligible_results;
+    for (const auto& result : results) {
+        int weight = coin_selection_params.tx_noinputs_size * WITNESS_SCALE_FACTOR;
+        for (const auto& coin : result.GetInputSet()) {
+            weight += CalculateMaximumSignedInputSize(coin.txout, &wallet, &coin_control) * WITNESS_SCALE_FACTOR;
```

Comment:
Can directly use `coin.input_bytes` instead of calculating the size again.

Reply:
Thanks, that's much better.